src/libxtp/settings.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ *            Copyright 2009-2020 The VOTCA Development Team
+ *                       (http://www.votca.org)
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License")
+ *
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *              http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <votca/xtp/settings.h>
+
+using votca::tools::Property;
+
+namespace votca {
+namespace xtp {
+
+void Settings::read_property(const Property& properties,
+                             const std::string& key) {
+  for (const auto& prop : properties.get(key)) {
+    const auto& name = prop.name();
+    this->_nodes[name] = prop;
+  }
+}
+
+void Settings::load_from_xml(const std::string& path) {
+  Property options;
+  options.LoadFromXML(path);
+  this->read_property(options, _root_key);
+}
+
+void Settings::merge(const Settings& other) {
+  // Merge general properties
+  for (const auto& pair : other._nodes) {
+    auto it = this->_nodes.find(pair.first);
+    if (it == this->_nodes.end()) {
+      const auto& val = pair.second;
+      this->_nodes[pair.first] = val;
+    }
+  }
+}
+
+bool Settings::exists(const std::string& key) const {
+  auto it = this->_nodes.find(key);
+  return (it != this->_nodes.end()) ? true : false;
+}
+
+Settings::Settings_map::const_iterator Settings::search_for_mandatory_keyword(
```

Comment:
why do you return a pointer? The name of the function does not make clear what it does.

Reply:
You are right. The method is now called `check_mandatory_keyword` and returns `void`

Reply:
why not a bool?

Reply:
Because we want to raise an error if the mandatory keyword is not available, see: https://github.com/votca/xtp/blob/settings/src/libxtp/settings.cc#L79